Gems/ROS2/Code/Source/Lidar/LidarSystem.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include "LidarSystem.h"
+#include "ROS2/Lidar/LidarRegistrarBus.h"
+
+namespace ROS2
+{
+    AzPhysics::SceneHandle GetPhysicsSceneFromEntityId(const AZ::EntityId& entityId)
+    {
+        auto* physicsSystem = AZ::Interface<AzPhysics::SystemInterface>::Get();
+        auto foundBody = physicsSystem->FindAttachedBodyHandleFromEntityId(entityId);
```

Comment:
When is the simulated body this is querying created?

It's important to think when this is asked and at what point they are created, otherwise this will always return invalid body.

From what I can see in the PR the chain of calls that arrive to here is as follows:

`ROS2LidarSensorComponent::Activate()` > `ConnectToLidarRaycaster()` > `CreateLidar(GetEntityId())` > `GetPhysicsSceneFromEntityId(entityId)` > FindAttachedBodyHandleFromEntityId(entityId)

So the body is expected to be already created when `ROS2LidarSensorComponent` is activated. 

When and where is the body created? Does it happen before  `ROS2LidarSensorComponent` is activated?

Reply:
Yes, I see where problems could arise now. I've moved this function into the LidarRaycaster.cpp file and made it so the handle is set upon first raycast (which guarantees that the entity with said ROS2LidarSensor component is activated). Thanks for spotting that!